### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by janvanrijn**

### CODE REVIEW

**Comment by janvanrijn** [comment]
This PR adds 'per_fold' functionality to evaluation list, and solves a bug in where runs can contain 'illegal' user defined measures

File: openml_OS/models/Estimation_procedure.php [hunk 1]
```diff
@@ -37,61 +37,76 @@ function get_by_parameters( $ttid, $type, $repeats, $folds, $percentage, $strati
...
-  public function trainingset_size( $datasetsize, $folds ) {
-    $foldsize = ceil($datasetsize / $folds);
-    return $foldsize * ($folds-1);
-  }
-  
-  public function number_of_samples( $trainingsetsize ) {
+  public function number_of_samples($trainingsetsize) {
     $i = 0;
-    for( ; $this->sample_size( $i, $trainingsetsize ) < $trainingsetsize; ++$i ) { }
+    for( ; $this->sample_size($i, $trainingsetsize) < $trainingsetsize; ++$i ) { }
```
Comment (joaquinvanschoren) [inline openml_OS/models/Estimation_procedure.php]: Weird construct. Why is $i defined outside of the loop?

File: openml_OS/models/Task.php [hunk 2]
```diff
@@ -69,7 +69,7 @@ function search( $task_type_id, $keyValues ) {
     return $this->query( $sql );
   }
   
-  // @deprecated, source of errors
+  // @deprecated, source of errors. REMOVE ASAP
```
Comment (joaquinvanschoren) [inline openml_OS/models/Task.php]: Can we remove this now?

File: openml_OS/models/api/v1/Api_evaluation.php [hunk 3]
```diff
@@ -127,53 +133,69 @@ private function evaluation_list($segs, $user_id) {
...
+          $where_runs;
         $count = $this->Evaluation->query($sql_test)[0]->count;
       } else {
         $sql_test =
           'SELECT count(distinct r.rid) as count ' .
           'FROM run r, task t, algorithm_setup s ' .
           'WHERE r.setup = s.sid AND r.task_id = t.task_id ' .
-          $where_runs .
-          $where_limit ;
+          $where_runs;
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_evaluation.php]: $where_limit is no longer used? Why not? if not necessary, remove entirely?

File: openml_OS/models/api/v1/Api_run.php [hunk 4]
```diff
@@ -510,15 +548,15 @@ private function run_upload() {
       $this->Run->outputData($runId, $did, 'runfile', $key);
     }
 
-    // attach input data
+    // attach input data TODO: JvR: this is legacy and implicit with the task. Remove?
```
Comment (joaquinvanschoren) [inline openml_OS/models/api/v1/Api_run.php]: No longer used? Remove?

File: openml_OS/views/pages/api_new/v1/xml/pre.php [hunk 5]
```diff
@@ -292,6 +293,8 @@
 $this->apiErrors[544] = 'Illegal filter specified';
 $this->apiErrors[545] = 'Can not specify offset without limit';
 $this->apiErrors[546] = 'Requested result limit too high. ';
+$this->apiErrors[547] = 'Per fold can only be set to value "true" or "false". ';
+$this->apiErrors[548] = 'Per fold queries are experimental and require a fair amount of filters on resulting run records to keep the query fast (use, e.g., flow, setup, task and uploader filter)';
```
Comment (joaquinvanschoren) [inline openml_OS/views/pages/api_new/v1/xml/pre.php]: unclear error message. You want to say 'you cannot use filter X without also using filter Y,Z'?

File: openml_OS/models/Estimation_procedure.php (hunk 1 above)
Comment (janvanrijn) [reply openml_OS/models/Estimation_procedure.php]: Looks like we want `$i` to have a broader scope. Was already the case, this line was not really changed.

File: openml_OS/models/Task.php (hunk 2 above)
Comment (janvanrijn) [reply openml_OS/models/Task.php]: I would be *highly* in favour of this, but the frontend task creation relies on this function.

File: openml_OS/models/api/v1/Api_evaluation.php (hunk 3 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_evaluation.php]: Good point. The logic of these statements seems a bit flawed

File: openml_OS/models/api/v1/Api_run.php (hunk 4 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_run.php]: legacy != no longer used. 

We should revise the db schema to remove this.

File: openml_OS/models/api/v1/Api_evaluation.php (hunk 3 above)
Comment (janvanrijn) [reply openml_OS/models/api/v1/Api_evaluation.php]: Nvm, I double checked an the logic seems fine.

**Pull request merged**